cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -103,6 +106,39 @@ func PerformPostUpgradeTasks(client clientset.Interface, cfg *kubeadmapi.InitCon
 		errs = append(errs, err)
 	}
 
+	if err := PerformAddonsUpgrade(client, cfg, out); err != nil {
+		errs = append(errs, err)
+	}
+
+	return errorsutil.NewAggregate(errs)
+}
+
+// PerformAddonsUpgrade performs the upgrade of the coredns and kube-proxy addons
+func PerformAddonsUpgrade(client clientset.Interface, cfg *kubeadmapi.InitConfiguration, out io.Writer) error {
```

Comment:
we should state it cleanly in the release note, that this will bring some different behavior after this features, the addon was upgraded immediately before 1.28 for the `kubeadm upgrade apply`, but now addons will only get updated after **all** control planes have been upgrade successfully if the FG is enabled.